src/vs/workbench/contrib/files/browser/fileActions.ts

```diff
@@ -155,13 +155,14 @@ export class GlobalNewUntitledFileAction extends Action {
 	constructor(
 		id: string,
 		label: string,
-		@IEditorService private readonly editorService: IEditorService
+		@ICommandService private readonly commandService: ICommandService // {{ SQL CARBON EDIT }} - commandService to open newQuery
 	) {
 		super(id, label);
 	}
 
 	run(): Promise<any> {
-		return this.editorService.openEditor({ options: { pinned: true } }); // untitled are always pinned
+		// {{ SQL CARBON EDIT }} - open newQuery
```

Comment:
Doesn't this break File -> New File though? It'd always open a sql query then.

Reply:
No, that goes through another action:

https://github.com/microsoft/azuredatastudio/blob/97b9c43c7a7aaad46be061d41eb2fa89db1e86a7/src/vs/workbench/contrib/files/browser/fileActions.ts#L146